### ISSUE TITLE
Report analysis progress messages in the app

### ISSUE THREAD

**Issue by adricaswell** [LandSciTech/ccviR#121]
Figure out how to report progress messages from spatial analysis seen in R in the app. "spatial analysis is complete" and messages about errors that stop the analysis already appear in app. This would give users using the app more insight into the spatial analysis progress, warnings, and errors.

**Comment by see24** [comment]
I would like to more generally improve the progress by using a progress bar and making the loading spinners work better for maps and results as well

**Comment by see24** [comment]
It would be nice if there was a distinction in the messages when spatial data is not provided for an optional data set vs for a required data set. Right now both have red text that reads like an error and users might think they have done something wrong for they ones where spatial data is optional. So for PTN I think the message should be black or grey and say "To complete the spatial analysis for this question provide a Physiological Thermal Niche polygon on the Spatial Data Analysis page. If spatial data is not available the question can be answered based on other information or left blank for unknown."

**Comment by steffilazerte** [comment]
What do you think of the following. It might be a bit colourful. I tried to minimize the text while keeping it clear, usable across pages (because section D groups questions 2 and 3), and consistent. 

![Image](https://github.com/user-attachments/assets/35722e2f-5810-4dd9-9720-868115ca07a0)

![Image](https://github.com/user-attachments/assets/88a4af62-3f6a-4703-a5f0-23f86ed7b332)

**Comment by see24** [comment]
I think it looks good overall. I think that technically the climate data package is allowed to not contain the data for the historical niches so it should be optional as well. Also I would replace "If no spatial," with "Spatial data not provided," It is a little repetitive but I think it is more clear

**Comment by see24** [comment]
In general the progress messages are much improved but the results page still needs some work. 
The page loads out of sequence with some of the text showing up before the figures. It would also be nice if there was some sort of spinner or progress bar for calculating the index, although it doesn't take to long so as long as the page loaded in a better order it would be fine

**Comment by steffilazerte** [comment]
Have you seen https://github.com/LandSciTech/ccviR/issues/155#issuecomment-2878101454 yet? I moved things around deliberately, but I wasn't sure if my logic made sense. Let me know what you think!

**Comment by see24** [comment]
Yes, I am happy with the data completeness part coming first. What I am wondering about is the loading of the results after the user clicks Calculate CCVI. Right now the text shows up about 5 seconds before the plots (this has been the case for a long time). It would be great if the text didn't display until the images were also ready.
